### PR TITLE
Add IPU workflow config actor

### PR DESCRIPTION
**Pull request by Rezney**
Also starts gathering LEAPP_DEVEL env vars.

### CODE REVIEW

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/library.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import os
+
+
+def get_env_vars():
+    """
+    Gather LEAPP_DEVEL environment variables and provide them as messages to be
+    available after reboot.
+    """
+    dev_vars = [(k, v) for (k, v) in os.environ.items() if k.startswith('LEAPP_DEVEL')]
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/library.py]:
```suggestion
    return [(k, v) for (k, v) in os.environ.items() if k.startswith('LEAPP_DEVEL')]
```

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/library.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    return dev_vars
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/library.py]:
```suggestion
```

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/actor.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    """
+
+    name = 'ipu_workflow_config'
+    consumes = ()
+    produces = (IPUConfig,)
+    tags = (IPUWorkflowTag,)
+
+    def process(self):
+        dev_env_vars = get_env_vars()
+        self.produce(IPUConfig(leapp_env_vars=dev_env_vars))
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/actor.py]:
```suggestion
        self.produce(IPUConfig(leapp_env_vars=get_env_vars()))
```

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/actor.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    IPU workflow config actor
+    """
+
+    name = 'ipu_workflow_config'
+    consumes = ()
+    produces = (IPUConfig,)
+    tags = (IPUWorkflowTag,)
+
+    def process(self):
+        dev_env_vars = get_env_vars()
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/actor.py]:
```suggestion
```

**Comment by vinzenz** [comment]
Somehow I'd go with all LEAPP_ environment variables, not only LEAPP_DEVEL,

File: repos/system_upgrade/el7toel8/models/ipuconfig.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+from leapp.topics import SystemInfoTopic
+
+
+class IPUConfig(Model):
+    """
+    IPU workflow configuration model
+    """
+    topic = SystemInfoTopic
+
+    leapp_env_vars = fields.List(fields.JSON(), default=[])
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/models/ipuconfig.py]: Why a list of JSON? This doesn't seem right. There are two things I can imagine:

`fields.List(fields.Model(EnvVar))`
or just `fields.JSON()`

Where EnvVar would be:

```python
class EnvVar(Model):
     topic = ...
     name = fields.String()
     value = fields.String()
```

Comment (vinzenz) [reply repos/system_upgrade/el7toel8/models/ipuconfig.py]: and to be straight, I prefer the List(EnvVar) approach

**Comment by Rezney** [comment]
@vinzenz I appreciate your checking this, however I just put this together fast (without much thinking) for testing.

I think it is good that people are (pre)checking also WIP PRs, so should we introduce something like "WIP-ignore"?

**Comment by vinzenz** [comment]
I didn't do a real 'review' I commented on things I saw :-) - This helps you on not spending your time on things that you might have to redo later on anyway. ;-)

**Comment by Rezney** [comment]
Ok, so we were both thinking about saving other side's time :) I will create a copr build manually next time then.

**Comment by pirat89** [comment]
Depends on Leapp PR: https://github.com/oamg/leapp/pull/505

**Comment by Rezney** [comment]
@vinzenz should I add ignore for "LEAPP_CURRENT_PHASE" and "LEAPP_CURRENT_ACTOR" or we do not care? Just asking it is minor thing.

**Comment by vinzenz** [comment]
@Rezney i guess they should be ignored there are some more we would have to add if necessary- just check what’s there for now

**Comment by Rezney** [comment]
@vinzenz There are these:

```
LEAPP_COMMON_TOOLS
LEAPP_CURRENT_ACTOR
LEAPP_HOSTNAME
LEAPP_VERBOSE
LEAPP_COMMON_FILES
LEAPP_EXECUTION_ID
LEAPP_CURRENT_PHASE
LEAPP_DEBUG
```

so we are ruling out "LEAPP_CURRENT_PHASE" and "LEAPP_CURRENT_ACTOR" for now.

**Comment by Rezney** [comment]
Added also a test...

**Comment by Rezney** [comment]
@vojtechsokol thanks for PR to this PR. I left a comment there.

**Comment by Rezney** [comment]
@vojtechsokol 's PR merged into this one...

**Pull request merged**